packages/esm-patient-registration-app/src/patient-registration/patient-registration-hooks.ts

```diff
@@ -180,49 +181,73 @@ export function useInitialFormValues(patientUuid: string): [FormValues, Dispatch
   return [initialFormValues, setInitialFormValues];
 }
 
-export function useInitialAddressFieldValues(patientUuid: string, fallback = {}): [object, Dispatch<object>] {
-  const { isLoading, patient } = usePatient(patientUuid);
+export function useInitialAddressFieldValues(
+  fallback = {},
+  isLoadingPatientToEdit: boolean,
+  patientToEdit: fhir.Patient,
+  patientUuid: string,
+): [object, Dispatch<object>] {
   const [initialAddressFieldValues, setInitialAddressFieldValues] = useState<object>(fallback);
 
   useEffect(() => {
     (async () => {
-      if (patient) {
-        setInitialAddressFieldValues({
-          ...initialAddressFieldValues,
-          address: getAddressFieldValuesFromFhirPatient(patient),
-        });
-      } else if (!isLoading && patientUuid) {
+      if (patientToEdit) {
+        setInitialAddressFieldValues((currentValues) => ({
+          ...currentValues,
+          address: getAddressFieldValuesFromFhirPatient(patientToEdit),
+        }));
+      } else if (!isLoadingPatientToEdit && patientUuid) {
         const registration = await getPatientRegistration(patientUuid);
         setInitialAddressFieldValues(registration?._patientRegistrationData.initialAddressFieldValues ?? fallback);
       }
     })();
-  }, [fallback, initialAddressFieldValues, isLoading, patient, patientUuid]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isLoadingPatientToEdit, patientToEdit, patientUuid]);
```

Comment:
I think we can avoid the need for this if we store the initial fallback as a ref and use the ref here? E.g.,

```ts
const fallbackRef = useRef(fallback);

// ...

?? fallbackRef.current
```

Same result, but we're obeying the rules of hooks better that way.